### PR TITLE
Add Support for Final Validation After Fine-Tuning

### PR DESCRIPTION
**Pull request by Geo99pro**
Hi everyone,  

I hope you're all doing well!  

First of all, I would like to congratulate you all on the amazing work you've done and made available to the community. It’s truly impressive!  

@JulioZhao97, I encountered an issue when trying to fine-tune the model on custom data. Specifically, the model was unable to perform the final validation step after training.  

This issue seems related to previous reports from other users. For example, the issue mentioned in [#72](https://github.com/opendatalab/DocLayout-YOLO/issues/72).  

After some debugging sessions, I made updates to the code to address this problem.  

**🆕 I've also added a Dockerfile, in case anyone wants to use it.**

I hope my contribution is helpful and adds value to your excellent project.  

Looking forward to your feedback!

Best regards,

### CODE REVIEW

**Comment by JulioZhao97** [comment]
@Geo99pro Hello! Thanks for your contribution, I will review your PR and give responses within this week (sorry for that cause I am kind of busy)

**Comment by Geo99pro** [comment]
> @Geo99pro Hello! Thanks for your contribution, I will review your PR and give responses within this week (sorry for that cause I am kind of busy)

Hi sir @JulioZhao97, There is no problem, i can understand it. I make myself available anytime.

File: doclayout_yolo/nn/modules/g2l_crm.py [hunk 1]
```diff
@@ -33,10 +33,15 @@ def __init__(self, c, dilation, k, fuse="sum", shortcut=True):
 
     def dilated_conv(self, x, dilation):
         act = self.dcv.act
-        bn = self.dcv.bn
         weight = self.dcv.conv.weight
         padding = dilation * (self.k//2)
-        return act(bn(F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation)))
+        if hasattr(self.dcv, "bn") and self.dcv.bn is not None:
```
Comment (Geo99pro) [inline doclayout_yolo/nn/modules/g2l_crm.py:38]: Sir @JulioZhao97 I did this because while debugging, I noticed that in: https://github.com/opendatalab/DocLayout-YOLO/blob/main/doclayout_yolo/nn/tasks.py more specifically in the fuse function (line 180) you merge the `Conv2d()` and the `BatchNorm2d()` whose purpose is to speed up inference. Since you delete after the bn, the model undergoes this alteration.

File: doclayout_yolo/nn/modules/g2l_crm.py [hunk 2]
```diff
@@ -33,10 +33,15 @@ def __init__(self, c, dilation, k, fuse="sum", shortcut=True):
...
+            bn = self.dcv.bn
+            x = bn(F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation))
+        else:
+            x = F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation)
+        
+        return act(x) # Apply activation function
```
Comment (Geo99pro) [inline doclayout_yolo/nn/modules/g2l_crm.py:44]: Sir @JulioZhao97 I did this because while debugging, I noticed that in: https://github.com/opendatalab/DocLayout-YOLO/blob/main/doclayout_yolo/nn/tasks.py more specifically in the ```fuse function (line 180)``` you merge the `Conv2d()` and the `BatchNorm2d()` whose purpose is to speed up inference. Since you delete after the bn, the model undergoes this alteration. I could be wrong, but it will be a pleasure to learn from you.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+stringzilla==3.12.3
+sympy==1.13.1
+thop==0.1.1.post2209072238
+torch==2.6.0
+torchvision==0.21.0
+tqdm==4.67.1
+triton==3.2.0
+typing-inspection==0.4.0
+typing_extensions==4.13.0
+tzdata==2025.2
```
Comment (Geo99pro) [inline requirements.txt:79]: Hello sir @JulioZhao97, I've added a requirements file to better manage the dependencies that will make it easy to use docker.

File: requirements.txt
Comment (Geo99pro) [inline requirements.txt:1]: Hello sir @JulioZhao97, I've added a requirements file to better manage the dependencies that will make it easy to use docker.
